### DISCUSSION TITLE
Predefined Base Card Templates

### DISCUSSION THREAD

**Discussion by hristoiankov** [oatear/cider#82] [Ideas]
When creating a new template, provide options for selecting a predefined base template.

Features:
- Include basic card sizes: Poker, Tarot, Square, Circle, Hexagon, etc.
- Include sub categories within those basic sizes that provide predefined template: Modern, Magic, Card Back, etc.
- Spend some time creatively coming up with card templates.
- Will need a global place to store template assets.
- ~Once template is part of the project, its assets should be added to the project assets (Ex. template-*).~
- Or there can be a set of built-in assets that can be pulled in to the cards (symbols, images, backs).

Built-in Assets:
- How will the assets be referenced? assets.cider.*
- How will the assets be displayed? toggle-show-cider-assets
- What assets will be available?
  - card backs
  - template fronts
  - symbols
  - banners
  - characters
  - landscapes
  - items

**Comment by jordanwallwork** [comment]
Is this being worked on? This would be really useful to me and I'd be keen to lend a hand with the development if I can be of any help?

**Reply by hristoiankov** [reply]
Probably the most useful thing here would be to have a survey of what sort of assets and templates users are looking for. I can then draw those out in order from most-requested to more-niche.

An alternative I've considered is collaborating with someone that has already created an asset pack and import that in.
Ex. https://graphicriver.net/card-graphics-in-game-assets

Some questions:
What kind of card game are most users creating (Ex. TCG, trick-taking)?
What is the most common theme (Ex. fantasy, simple/uno)?
What are the most common symbols people use (Ex. coin, life, etc.)?

**Reply by jordanwallwork** [reply]
The bit that would be really useful to me isn't the predefined templates so much, but the ability to specify additional card settings. I kinda focused on the "Include basic card sizes: Poker, Tarot, Square, Circle, Hexagon, etc." bit and glossed over the rest lol.

I'm playing around with CIDEr for a project currently and it uses landscape cards; I know I could fake this by applying `transform: rotate(90deg)` to the card contents, but it makes everything more confusing. Being able to specify "this is a landscape card, tarot size" would be great.

**Reply by hristoiankov** [reply]
You don't need to rotate to get a landscape card design, you can just switch the width and height values in the CSS of the template.
![image](https://github.com/oatear/cider/assets/35353708/e464825a-a009-4d67-a6d1-d7f4d73695f4)

**Reply by jordanwallwork** [reply]
Oh my god I can't believe I missed that 🤦 thanks!
